### PR TITLE
[Solution] Fix more code/compiler warnings

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs [hunk 1]
```diff
@@ -93,20 +92,18 @@ public IActionResult GetAddressIndexerTip()
...
-            }
 
             try
             {
                 Block block = this.blockStore.GetBlock(uint256.Parse(query.Hash));
 
                 if (block == null)
                 {
-                    return new NotFoundObjectResult("Block not found");
+                    return Ok("Block not found");
```
Comment (MithrilMan) [inline src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs]: why this change?

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs]: `NotFoundObjectResult` creates 404 status codes, which relates to whether or not the caller could call the API end point. In this instance we did in fact call the endpoint successfully but didn't find the block, so the HTTP call is still valid.

File: src/Stratis.Bitcoin.Features.Consensus/CoinviewPrefetcher.cs [hunk 2]
```diff
@@ -42,17 +47,19 @@ public CoinviewPrefetcher(ICoinView coinview, ChainIndexer chainIndexer, ILogger
...
-        /// pre-fetching is done on the background.
+        /// Pre-fetch UTXOs for a block and some blocks with higher height.
+        /// <para>
+        /// This task runs in the background.
+        /// </para>
         /// </summary>
-        /// <param name="header">Header of a block that is about to be partially validated and requires pre-fetching.</param>
-        private async Task OnHeaderEnqueuedAsync(ChainedHeader header, CancellationToken cancellation)
+        /// <param name="header">Header of a block that is about to be partially validated and that requires pre-fetching.</param>
+        private Task OnHeaderEnqueued(ChainedHeader header, CancellationToken cancellation)
```
Comment (MithrilMan) [inline src/Stratis.Bitcoin.Features.Consensus/CoinviewPrefetcher.cs]: in this kind of scenarios we may want to use ValueTask instead
see https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/

File: src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs (hunk 1 above)
Comment (MithrilMan) [reply src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs]: but AFAIK this is the standard behavior of REST APIs, so if we want to adhere to that standard, 404 is used to communicate that a resource is not found.
Basically in this case, if you ask for a block and it can't be find, it's not correct to return a successful message (status 200) 
e.g. take a look at this response https://stackoverflow.com/a/9930716/1716620

Consider that returning 404 is fine and we can add a content to that request that would be helpful to understand if the 404 was because of a failure on endpoint or because of a "block not found"

Comment (MithrilMan) [reply src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs]: as use case, even twitter uses 404
https://developer.twitter.com/en/docs/basics/response-codes.html

**Comment by fassadlr** [comment]
https://dev.azure.com/Stratisplatformuk/StratisBitcoinFullNode/_workitems/edit/4070

File: src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs [hunk 3]
```diff
@@ -278,24 +276,18 @@ public object GetBlockHeader(string hash, bool isJsonFormat = true)
...
 
-            return null;
+            BlockHeader blockHeader = this.ChainIndexer.GetHeader(uint256.Parse(hash))?.Header;
+
+            if (blockHeader == null)
+                return null;
+
+            if (isJsonFormat)
+                return new BlockHeaderModel(blockHeader);
+            else
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs]: Hmm, is this `else` superfluous now?

File: src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs [hunk 4]
```diff
@@ -457,6 +460,10 @@ private bool ProcessBlock(Block block, ChainedHeader header)
         }
 
         /// <summary>Adds a new balance change entry to to the <see cref="addressIndexRepository"/>.</summary>
+        /// <param name="height">The height of the block this being processed.</param>
+        /// <param name="address">The address receiving the funds.</param>
+        /// <param name="amount">The amount being received.</param>
+        /// <param name="deposited"><c>true</c> if this is an output being spent, <c>false</c> otherwise.</param>
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs]: I think this might be backwards, a deposit is an incoming transaction for an address?

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs]: @zeptin it seems we pass false here when call process from an TxInput and true from an TxOutput?

Comment (zeptin) [reply src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs]: Yes, but bear in mind that when we see a TxIn for a given scriptPubkey (address), that means that the prevOut in question is getting spent. So it is a withdrawal 'from' the address and therefore not a deposit. Conversely with the TxOut.

File: src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs (hunk 3 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs]: Fixed :)

File: src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs (hunk 4 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs]: OK fixed :) please check 👍

**Comment by fassadlr** [comment]
I have synced main and test net and were still able to stake.
